source/ocs/helper.ts

```diff
@@ -0,0 +1,52 @@
+import { OcsSharePermissions } from './types';
+import { OcsError }            from '../errors';
+
+export function rejectWithOcsError(
+  message: string,
+  reason: string,
+  identifier?: string | number,
+  statusCode?: string
+) {
+  const errorObj = {
+    message,
+    reason
+  };
+
+  assignDefined(errorObj, {
+    identifier,
+    statusCode
+  });
+
+  return Promise.reject(new OcsError(errorObj));
+}
+
+export function assignDefined(target, ...sources) {
+  for (const source of sources) {
+    for (const key of Object.keys(source)) {
+      const val = source[key];
+      if (val !== undefined) {
+        target[key] = val;
+      }
+    }
+  }
+}
+
+export function ocsSharePermissionsToText(permissions: OcsSharePermissions) : string {
```

Comment:
What about adding this as a `toString` method on the `OcsSharePermissions` class?

Reply:
OcsSharePermissions is an enum. This method returns a human-readable string representation of the enabled enum flags.

Reply:
Alright, you got me… I didn't check what `OcsSharePermissions` actually has as a type.